### ISSUE TITLE
Tab Session Manager Integration

### ISSUE THREAD

**Issue by prizrak9** [mbnuqw/sidebery#59]
I use Tab Session Manager extension and each time I restore tabs I see my trees broken. Some integration would be really nice.

**Comment by br4nnigan** [comment]
well I guess you have to use the snapshot feature within sidebery

**Comment by onguarde** [comment]
@dexeonify thanks that works. but the ordering of tabs gets messed up. Do you face the same issue?

If you wish to continue using the built-in snapshot feature, you can setup a shortcut to make creating snapshots easier. Either a keyboard shortcut, or add a button.

![image](https://user-images.githubusercontent.com/681355/226303676-fdc2d911-496e-4772-a7d3-bbd7d38e30c9.png)

![image](https://user-images.githubusercontent.com/681355/226297576-425e180d-77e1-45b3-8b42-be7349e7f39f.png)

I've also created a bookmark pointing to the "view snapshots" section in sideberry's settings. 

Then, you can optionally configure a keyword for that bookmark for that page, this allows you to quickly jump to it, by typing a few letters in the omnibar. 

![image](https://user-images.githubusercontent.com/681355/226298466-92e6d7c9-25db-4f77-9071-3a459cc390da.png)

**Comment by onguarde** [comment]
With a few enhancements above, we don't need custom extensions to support session restore. I propose some work go into reducing the number of clicks in creating and managing snapshots, without jumping through hoops to configure it, per my post above. 

Quite happy with the above setup. Shortcut to quickly create snapshots. Bookmark to quickly manage snapshots.

**Comment by dexeonify** [comment]
> thanks that works. but the ordering of tabs gets messed up. Do you face the same issue?

I haven't used Tab Session Manager in a while (I also haven't experience any unfortunate tab loss events lately 🙂), so I can't comment on that. Though a simple test _now_ shows that it restored the tabs perfectly.

> ![image](https://user-images.githubusercontent.com/681355/226303676-fdc2d911-496e-4772-a7d3-bbd7d38e30c9.png)

By the way, may I ask how did you get the "Create tabs panel" icon to be on the left? This is my configuration on a fresh profile and fresh Sidebery install, but no matter what I try the icon will be on the right.
![image](https://user-images.githubusercontent.com/72267349/226384196-7dd5bbb9-c48a-444d-8705-b98d00689b64.png)
![image](https://user-images.githubusercontent.com/72267349/226384463-3e37c834-6e58-4115-a010-e686bea17e5c.png)

**Comment by good-nitrogen** [comment]
> @dexeonify thanks that works. but the ordering of tabs gets messed up. Do you face the same issue?
> 
> If you wish to continue using the built-in snapshot feature, you can setup a shortcut to make creating snapshots easier. Either a keyboard shortcut, or add a button.
> 
> ![image](https://user-images.githubusercontent.com/681355/226303676-fdc2d911-496e-4772-a7d3-bbd7d38e30c9.png)
> 
> ![image](https://user-images.githubusercontent.com/681355/226297576-425e180d-77e1-45b3-8b42-be7349e7f39f.png)
> 
> I've also created a bookmark pointing to the "view snapshots" section in sideberry's settings.
> 
> Then, you can optionally configure a keyword for that bookmark for that page, this allows you to quickly jump to it, by typing a few letters in the omnibar.
> 
> ![image](https://user-images.githubusercontent.com/681355/226298466-92e6d7c9-25db-4f77-9071-3a459cc390da.png)

How did you add the snapshot/search buttons and move the bookmark to that side also? I haven't figured it out yet

**Comment by albino1** [comment]
> How did you add the snapshot/search buttons and move the bookmark to that side also? I haven't figured it out yet

`Settings` -> `Navigation Bar` -> `Enabled Elements`

Then just drag `Create Snapshot` into Enabled and position it where you want.

If you want something to be on the left, put it before the `Space` in the order.

**Comment by dexeonify** [comment]
The thing is, if you've seen my screenshot above, it doesn't seem to work. Weirdly enough the bookmark element is already on the left for me by default, and again, I can't move it to the right whatever I try.

**Comment by good-nitrogen** [comment]
> > How did you add the snapshot/search buttons and move the bookmark to that side also? I haven't figured it out yet
> 
> `Settings` -> `Navigation Bar` -> `Enabled Elements`
> 
> Then just drag `Create Snapshot` into Enabled and position it where you want.
> 
> If you want something to be on the left, put it before the `Space` in the order.

"Enabled elements" isn't an option in my sidebery settings. I'm using firefox on windows 11 if that helps
![image](https://user-images.githubusercontent.com/67857202/235401328-d44834e4-d04d-4f8c-8ddb-e4786de811a2.png)

**Comment by albino1** [comment]
It looks like you're on Sidebery 4, most people on this github are using Sidebery 5 beta:

https://github.com/mbnuqw/sidebery/releases/tag/v5.0.0b31

**Comment by good-nitrogen** [comment]
Oh my goodness I see, thank you so much I think I got it now!!

**Comment by Polisson** [comment]
> Did you enable "Supports Tree Style Tab" in TSM settings? I've set the delay to as low as 10ms and it always restore the tab tree correctly. Unfortunately it doesn't support restoring panels.

Yes, that unfortunately seems to be correct at my setup - my tabs are messed up according to panels. 

I don't use TSM but really looking for such a solution to continuous browse on three different machines with *fox browsers - saying when I close the browser on one browser and open on the second machine I would continue with the same tree view. 
TSM was the first shot, but if it is incompatible with Sidebery, I'm looking other ways to workaround. So second is Sidebery snapshots, but I have to sync them on my side (SyncThing), but...
TSM has unique features like:
- syncing to GDrive, 
- support to TST (Sidebery competition)
- saving session on browser close
- replace the current session with a saved one or add saved to current (in same window)

Having TSM work correctly with Sidebery, would be nice. But maybe Sidebery snapshot syncing is the proper way to achieve it.
